README.md

```diff
@@ -27,20 +27,64 @@ application, and allows for the user to then do any further setup required once
-the original application used. The next assumption is that DEFAULTS_FILE is
-specified in settings.cfg, or as an environment variable.  This file is already
-completely flexible, and can contain any additional configuration your
-application may need.  It is assumed this is a yaml file.  The contents of
-the file are parsed and stored in app.config.default_params.
+The library supports two levels of configuration.
+
+#### Level 1: Flask Settings
+
+The first is the standard flask configuration that can be used by default, but
+with a bit of extra structure.  You can specify this configuration using any or
+all of the following options:
+
+1. Pass in the Flask config to AppScaffold via the config parameter (this is a
+   dictionary).
+2. Via a standard flask settings.cfg file.  Flask will look for this in the
+   instance folder, which you can specify via the instance_path parameter to
+   AppScaffold if it is not in the default location ('instance' within the app).
+3. Via a FLASK_SETTINGS environment variable whose value is a path to a valid
+   Flask settings.cfg file. This can be a relative path if the instance_config
+   folder is specified, or can be an absolute path in all cases.
+
+Note that Flask requires all config settings to be in CAPS, otherwise they will
+not be included in the app.config dictionary on initialization.
+
+AppScaffold will look for each of the items above, and they will be set in the
+same order, if found.  So, for example, if you set:
+
+    config= {'FOO': 'bar'}
+
+when you call AppScaffold, but then have:
+
+    FOO='something else'
+
+in your file specified by the FLASK_SETTINGS environment variable, the latter
+will overwrite the former.
+
+#### Level 2: Custom Settings
+
+Custom settings are meant to be more flexible than the Flask settings, and can
+be in whatever structure makes sense for your application.  These settings are
+found and loaded by AppScaffold when you reference a Flask setting of
+CUSTOM_SETTINGS in any of the following ways:
+
+1. As a key in your config dict passed into AppScaffold
+2. As a key in your settings.cfg file
+3. As an environment variable whose value is a path to a valid file containing
+   your custom configuration. This can be a relative path if the
+   instance_config folder is specified, or can be an absolute path in all cases.
+
+Currently, settings can be configured via a standard cfg file (using
+[ini-file](https://docs.python.org/3/library/configparser.html#supported-ini-file-structure)
+format) or a yaml file (which can end with '.yml' or '.yaml'). These files, in
+turn, can reference additional files if needed.  Sections and structures are
+supported, so long as they can be put into a python dictionary, and will be
+added as-is, without additional formatting of case (which the python
+ConfigParser library does by default). Also, keys can be in whatever case suits
+your needs, which is a difference from the core Flask settings.
+
+### Logger Formatting
```

Comment:
I like the readme additions, they are very helpful! One thing I would suggest for a follow-on patch is to add some usage examples to make it clear how to use this project. We could even add an example directory that shows each of these 3 levels and how they impact your flask config or some general config settings you want added.

Reply:
Absolutely, I expect to refine this over the next few patches, and this should definitely be on the todo list.  I'll create an issue to track it, so I don't forget.

Reply:
Awesome, thanks.